### ISSUE TITLE
Service chaining timeout issue for get request

### ISSUE THREAD

**Issue by madushadhanushka** [wso2/product-ei#5220]
**Description:**
Buffer timeout error occur while trying to sending request to a second backend after getting large response from the first backend. Here it try to send response from the first backend to the second backend and since it's payload large it try to flush the buffer and wait until buffer get consumed.

**Steps to Produce**
Use the following API which has first endpoint that generates a large response (~60KB).
```
<api xmlns="http://ws.apache.org/ns/synapse" name="testapi" context="/testapi" version="1" version-type="context">
   <resource methods="GET">
      <inSequence>
         <log level="custom">
            <property name="where" value="=======starting Buffertest_API api====="/>
         </log>
         <log level="custom">
            <property name="where" value="=======calling EagleEye_GetWalletAccountsByWalletId_EP====="/>
         </log>
         <property name="Content-Type" value="application/json" scope="axis2" type="STRING"/>
         <property name="Accept" value="application/json" scope="transport" type="STRING"/>
         <property name="DISABLE_CHUNKING" value="true" scope="axis2" type="STRING"/>
         <header name="X-EES-AUTH-CLIENT-ID" scope="transport" expression="$ctx:clientId"/>
         <header name="X-EES-AUTH-HASH" scope="transport" expression="$ctx:HASHED_VALUE"/>
         <call>
            <endpoint>
               <http method="GET" uri-template="http://run.mocky.io/v3/c9886782-66b7-4f5e-8f43-108f749d6187"/>
            </endpoint>
         </call>
         <log level="custom">
            <property name="where" value="=======calling EagleEye_GetWalletByIdentityValue_EP ====="/>
         </log>
         <property name="NO_ENTITY_BODY" scope="axis2" action="remove"/>
         <property name="Content-Type" value="application/json" scope="axis2" type="STRING"/>
         <property name="Accept" value="application/json" scope="transport" type="STRING"/>
         <property name="DISABLE_CHUNKING" value="true" scope="axis2" type="STRING"/>
         <header name="X-EES-AUTH-CLIENT-ID" scope="transport" expression="$ctx:clientId"/>
         <header name="X-EES-AUTH-HASH" scope="transport" expression="$ctx:HASHED_VALUE"/>
         <call>
            <endpoint>
               <http method="GET" uri-template="http://run.mocky.io/v3/1cd23431-d538-4bf1-8995-36d7726bc9ec"/>
            </endpoint>
         </call>
         <log>
            <property name="where" value="-------------sent request to backend=======--------"/>
         </log>
         <send/>
      </inSequence>
   </resource>
</api>
```
When this API invoke it send the response to the client. But after socket timeout 3min it generate following exception.
```
[2020-07-10 19:32:37,778] [EI-Core] ERROR - PassThroughHttpSender IO while building message
org.apache.axiom.ext.io.StreamCopyException: Error writing to destination
	at org.apache.axiom.util.blob.OverflowBlob.writeTo(OverflowBlob.java:437)
	at org.apache.synapse.transport.passthru.PassThroughHttpSender.sendRequestContent(PassThroughHttpSender.java:386)
	at org.apache.synapse.transport.passthru.PassThroughHttpSender.invoke(PassThroughHttpSender.java:277)
	at org.apache.axis2.engine.AxisEngine.send(AxisEngine.java:442)
	at org.apache.synapse.core.axis2.DynamicAxisOperation$DynamicOperationClient.send(DynamicAxisOperation.java:185)
	at org.apache.synapse.core.axis2.DynamicAxisOperation$DynamicOperationClient.executeImpl(DynamicAxisOperation.java:167)
	at org.apache.axis2.client.OperationClient.execute(OperationClient.java:149)
	at org.apache.synapse.core.axis2.Axis2FlexibleMEPClient.send(Axis2FlexibleMEPClient.java:634)
	at org.apache.synapse.core.axis2.Axis2Sender.sendOn(Axis2Sender.java:85)
	at org.apache.synapse.core.axis2.Axis2SynapseEnvironment.send(Axis2SynapseEnvironment.java:569)
	at org.apache.synapse.endpoints.AbstractEndpoint.send(AbstractEndpoint.java:398)
	at org.apache.synapse.endpoints.HTTPEndpoint.send(HTTPEndpoint.java:85)
	at org.apache.synapse.endpoints.IndirectEndpoint.send(IndirectEndpoint.java:55)
	at org.apache.synapse.mediators.builtin.CallMediator.handleNonBlockingCall(CallMediator.java:278)
	at org.apache.synapse.mediators.builtin.CallMediator.mediate(CallMediator.java:122)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:109)
	at org.apache.synapse.mediators.base.SequenceMediator.mediate(SequenceMediator.java:260)
	at org.apache.synapse.core.axis2.Axis2SynapseEnvironment.mediateFromContinuationStateStack(Axis2SynapseEnvironment.java:805)
	at org.apache.synapse.core.axis2.Axis2SynapseEnvironment.injectMessage(Axis2SynapseEnvironment.java:303)
	at org.apache.synapse.core.axis2.SynapseCallbackReceiver.handleMessage(SynapseCallbackReceiver.java:578)
	at org.apache.synapse.core.axis2.SynapseCallbackReceiver.receive(SynapseCallbackReceiver.java:195)
	at org.apache.axis2.engine.AxisEngine.receive(AxisEngine.java:180)
	at org.apache.synapse.transport.passthru.ClientWorker.run(ClientWorker.java:265)
	at org.apache.axis2.transport.base.threads.NativeWorkerPool$1.run(NativeWorkerPool.java:172)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Output buffer write time out exceeded
	at org.apache.synapse.transport.passthru.Pipe$ByteBufferOutputStream.write(Pipe.java:583)
	at java.io.OutputStream.write(OutputStream.java:75)
	at org.apache.axiom.util.blob.OverflowBlob.writeTo(OverflowBlob.java:431)
	... 26 more
[2020-07-10 19:32:37,781] [EI-Core] ERROR - Axis2Sender Unexpected error during sending message out
org.apache.axis2.AxisFault: IO while building message
	at org.apache.synapse.transport.passthru.PassThroughHttpSender.handleException(PassThroughHttpSender.java:688)
	at org.apache.synapse.transport.passthru.PassThroughHttpSender.sendRequestContent(PassThroughHttpSender.java:404)
	at org.apache.synapse.transport.passthru.PassThroughHttpSender.invoke(PassThroughHttpSender.java:277)
	at org.apache.axis2.engine.AxisEngine.send(AxisEngine.java:442)
	at org.apache.synapse.core.axis2.DynamicAxisOperation$DynamicOperationClient.send(DynamicAxisOperation.java:185)
	at org.apache.synapse.core.axis2.DynamicAxisOperation$DynamicOperationClient.executeImpl(DynamicAxisOperation.java:167)
	at org.apache.axis2.client.OperationClient.execute(OperationClient.java:149)
	at org.apache.synapse.core.axis2.Axis2FlexibleMEPClient.send(Axis2FlexibleMEPClient.java:634)
	at org.apache.synapse.core.axis2.Axis2Sender.sendOn(Axis2Sender.java:85)
	at org.apache.synapse.core.axis2.Axis2SynapseEnvironment.send(Axis2SynapseEnvironment.java:569)
	at org.apache.synapse.endpoints.AbstractEndpoint.send(AbstractEndpoint.java:398)
	at org.apache.synapse.endpoints.HTTPEndpoint.send(HTTPEndpoint.java:85)
	at org.apache.synapse.endpoints.IndirectEndpoint.send(IndirectEndpoint.java:55)
	at org.apache.synapse.mediators.builtin.CallMediator.handleNonBlockingCall(CallMediator.java:278)
	at org.apache.synapse.mediators.builtin.CallMediator.mediate(CallMediator.java:122)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:109)
	at org.apache.synapse.mediators.base.SequenceMediator.mediate(SequenceMediator.java:260)
	at org.apache.synapse.core.axis2.Axis2SynapseEnvironment.mediateFromContinuationStateStack(Axis2SynapseEnvironment.java:805)
	at org.apache.synapse.core.axis2.Axis2SynapseEnvironment.injectMessage(Axis2SynapseEnvironment.java:303)
	at org.apache.synapse.core.axis2.SynapseCallbackReceiver.handleMessage(SynapseCallbackReceiver.java:578)
	at org.apache.synapse.core.axis2.SynapseCallbackReceiver.receive(SynapseCallbackReceiver.java:195)
	at org.apache.axis2.engine.AxisEngine.receive(AxisEngine.java:180)
	at org.apache.synapse.transport.passthru.ClientWorker.run(ClientWorker.java:265)
	at org.apache.axis2.transport.base.threads.NativeWorkerPool$1.run(NativeWorkerPool.java:172)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.axiom.ext.io.StreamCopyException: Error writing to destination
	at org.apache.axiom.util.blob.OverflowBlob.writeTo(OverflowBlob.java:437)
	at org.apache.synapse.transport.passthru.PassThroughHttpSender.sendRequestContent(PassThroughHttpSender.java:386)
	... 25 more
Caused by: java.io.IOException: Output buffer write time out exceeded
	at org.apache.synapse.transport.passthru.Pipe$ByteBufferOutputStream.write(Pipe.java:583)
	at java.io.OutputStream.write(OutputStream.java:75)
	at org.apache.axiom.util.blob.OverflowBlob.writeTo(OverflowBlob.java:431)
	... 26 more
```

**Comment by madushadhanushka** [comment]
Fixed with https://github.com/wso2/wso2-synapse/pull/1623

**Comment by yasirjanjua** [comment]
@madushadhanushka is this fix available in the GA release of micro-integrator 1.7?
